src/c_lib/base/base_model.pyx

```diff
@@ -150,7 +151,7 @@ def core_simulator(method,
 # _____________________________________________________________________________
 
 # frequency contrib
-    cdef ndarray[bool_t] f_contrib = np.asarray(freq_contrib, dtype=bool)
+    cdef ndarray[cnp.uint8_t] f_contrib = np.asarray(freq_contrib, dtype=np.uint8)

```

Comment:
Can you add a comment here that `unsigned __int8` is the same as `unsigned char`? It would be useful to have when visiting this code later.

Reply:
Added